src/App.js

```diff
@@ -14,11 +16,109 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    preparedGoods: [],
+    didAlreadyStart: false,
+  }
+
+  start = () => {
+    this.setState({
+      preparedGoods: [...goodsFromServer],
+      didAlreadyStart: true,
+    });
+  }
+
+  reverse = () => {
+    this.setState(prevState => (
+      {
+        preparedGoods: [...prevState.preparedGoods].reverse(),
+      }
+    ));
+  }
+
+  sortAlphabetically = () => {
+    this.setState({
+      preparedGoods: [...goodsFromServer].sort(),
+    });
+  }
+
+  reset = () => {
+    this.setState({
+      preparedGoods: [...goodsFromServer],
+    });
+  }
+
+  sortByLength = () => {
+    this.setState(prevState => ({
+      preparedGoods: [...prevState.preparedGoods]
+        .sort((a, b) => a.length - b.length),
```

Comment:
your method doesn't work properly, you have 8 letters in 'Ice cream' and 9 letters in 'Dumplings', you need to remove the space between words and then it'll work correctly

Reply:
![image](https://user-images.githubusercontent.com/41700032/92237403-f8e2d180-eebf-11ea-9871-c60a64c2b4c9.png)
there are not details about whitespaces. I had understood it like full length